bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericBluetoothHandler.java

```diff
@@ -161,9 +179,11 @@ private void updateThingChannels() {
                     logger.trace("{} processing characteristic {}", address, characteristic.getUuid());
                     CharacteristicHandler handler = getCharacteristicHandler(characteristic);
                     List<Channel> chans = handler.buildChannels();
+                    List<ChannelUID> chanUids = chans.stream().map(Channel::getUID).collect(Collectors.toList());
                     for (Channel channel : chans) {
                         channelHandlers.put(channel.getUID(), handler);
                     }
```

Comment:
Just a minor optimization.
```suggestion
                    for (ChannelUID channel : chanUids) {
                        channelHandlers.put(channel, handler);
                    }
```